features/admin.users.v1/components/user-change-password.tsx

```diff
@@ -529,25 +545,23 @@ export const ChangePasswordComponent: FunctionComponent<ChangePasswordPropsInter
                             type="password"
                             value=""
                             listen={ (values: Map<string, FormValue>): void => {
-                                if (values?.get("newPassword") === values?.get("confirmPassword")
-                                    && values?.get("confirmPassword") !== "") {
-                                    setIsConfirmPasswordMatch(true);
-
-                                    return;
-                                }
+                                const newPassword: string | undefined = values?.get("newPassword")?.toString();
+                                const confirmPassword: string | undefined = values?.get("confirmPassword")?.toString();
 
-                                if (isConfirmPasswordMatch !== undefined) {
-                                    setIsConfirmPasswordMatch(undefined);
-                                }
+                                setIsConfirmPasswordMatch(checkPasswordsMatch(newPassword, confirmPassword));
                             } }
                             validation={
                                 (value: string, validation: Validation, formValues: Map<string, FormValue>) => {
-                                    if (formValues?.get("newPassword") !== value) {
+                                    const newPassword: string | undefined = formValues?.get("newPassword")?.toString();
+
+                                    if (newPassword !== value) {
                                         validation.isValid = false;
                                         setIsConfirmPasswordMatch(false);
                                         validation?.errorMessages?.push(
                                             t("user:forms.addUserForm.inputs" +
                                             ".confirmPassword.validations.mismatch"));
+                                    } else if (value) {
+                                        setIsConfirmPasswordMatch(true);
                                     }
```

Comment:
In the current way this logic is written, it implicitly assumes that if value exists and isn’t mismatched, then it's valid. This might not be obvious at first glance.It can confuse readers because value was already involved in the previous condition, so it's unclear why it's being checked again.

How about if. we refactor the logic as follows for better readability? WDYT?

```suggestion
                                    if (value && newPassword !== value) {
                                        validation.isValid = false;
                                        setIsConfirmPasswordMatch(false);
                                        validation?.errorMessages?.push(
                                            t("user:forms.addUserForm.inputs" +
                                            ".confirmPassword.validations.mismatch"));
                                    } else {
                                        setIsConfirmPasswordMatch(true);
                                    }
```